code/game/gamemodes/dynamic/dynamic_rulesets_midround.dm

```diff
@@ -869,3 +869,56 @@
 		log_game("DYNAMIC: Not enough players volunteered for the ruleset [name] - [candidates.len] out of [required_candidates].")
 		return FALSE
 	return TRUE
+
+//////////////////////////////////////////////
+//                                          //
+//           NINJA      (GHOST)             //
+//                                          //
+//////////////////////////////////////////////
+
+/datum/dynamic_ruleset/midround/from_ghosts/ninja
+	name = "Space Ninja"
+	midround_ruleset_style = MIDROUND_RULESET_STYLE_HEAVY
+	antag_flag = ROLE_NINJA
+	required_type = /mob/dead/observer
+	antag_datum = /datum/antagonist/ninja
+	enemy_roles = list(JOB_NAME_SECURITYOFFICER, JOB_NAME_DETECTIVE, JOB_NAME_WARDEN, JOB_NAME_HEADOFSECURITY, JOB_NAME_CAPTAIN)
+	required_enemies = list(2,2,2,2,2,2,2,2,2,2)
+	required_candidates = 1
+	weight = 3
+	cost = 9
+	minimum_players = 20
+	repeatable = TRUE
+	var/spawn_loc
+
+/datum/dynamic_ruleset/midround/from_ghosts/ninja/execute()
+	//selecting a spawn_loc
+	if(!spawn_loc)
+		var/list/spawn_locs = list()
+		for(var/obj/effect/landmark/carpspawn/L in GLOB.landmarks_list)
+			if(isturf(L.loc))
+				spawn_locs += L.loc
+		if(!spawn_locs.len)
+			return FALSE
+		spawn_loc = pick(spawn_locs)
+	if(!spawn_loc)
+		return FALSE
+	return ..()
+
+/datum/dynamic_ruleset/midround/from_ghosts/ninja/generate_ruleset_body(mob/applicant)
+	//spawn the ninja and assign the candidate
+	var/mob/living/carbon/human/Ninja = create_space_ninja(spawn_loc)
+
+	//Prepare ninja player mind
+	var/datum/mind/Mind = new /datum/mind(applicant.key)
+	Mind.assigned_role = ROLE_NINJA
+	Mind.special_role = ROLE_NINJA
```

Comment:
this entire proc can be removed and simplified to a `generate_ruleset_body` to adjust the ninja's name etc. Everything else is typical and handled by the parent function already.

Reply:
generate_ruleset_body will use the clients preferences which we don't want, which means even if we called ..() we would have to undo all of the work done in the parent proc making this proc the same anyway.

Reply:
Ah, I see what you mean now. finish_setup handles the application of antagonist roles